src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,90 @@
 import React from 'react';
 import './TodoList.scss';
+import PropTypes from 'prop-types';
+import { Todo } from '../Todo';
+import { TodoSearch } from '../TodoSearch/TodoSearch';
+import { StatusSelect } from '../StatusSelect/StatusSelect';
 
-export const TodoList = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+export class TodoList extends React.Component {
+  state = {
+    query: '',
+    selectedOption: '',
+  }
+
+  searchHandler = (event) => {
+    const { value } = event.target;
+
+    this.setState({
+      query: value.toLowerCase(),
+    });
+  }
+
+  handleSelect = (event) => {
+    const { value } = event.target;
+
+    this.setState({
+      selectedOption: value,
+    });
+  }
+
+  render() {
+    const { query, selectedOption } = this.state;
+    const { selectedTodoId, handleChange, todos } = this.props;
+
+    const searchableTodos = todos.filter(({ title }) => (
+      title.toLowerCase().includes(query)
+    ));
+
+    const selectedTodos = searchableTodos.filter((todo) => {
+      if (selectedOption === 'active') {
+        return todo.completed === false;
+      }
+
+      if (selectedOption === 'completed') {
+        return todo.completed === true;
+      }
+
+      return true;
+    });
```

Comment:
```suggestion
    let selectedTodos = '';

    if (selectedStatus === '') {
      selectedTodos = searchableTodos;
    } else {
      selectedTodos = searchableTodos.filter((todo) => {
        if (selectedStatus === 'active') {
          return todo.completed === false;
        }

        if (selectedStatus === 'completed') {
          return todo.completed === true;
        }

        return true;
      });
    }
```
Чтоб лишний раз не запускался фильтр, если не выбран статус в селекторе, реализовал через `if/else`. Но мне это кажется теперь слишком громоздким. Норм?